src/net.cpp

```diff
@@ -1391,6 +1391,13 @@ void CConnman::NotifyNumConnectionsChanged()
         LOCK(cs_vNodes);
         vNodesSize = vNodes.size();
     }
+
+    // If we had zero connections before and new connections now or if we just dropped
+    // to zero connections reset the sync process if its outdated.
+    if (vNodesSize && !nPrevNodeCount || !vNodesSize && nPrevNodeCount) {
```

Comment:
Let's avoid implicit conversions here
```suggestion
    if ((vNodesSize > 0 && nPrevNodeCount == 0) || (vNodesSize == 0 && nPrevNodeCount > 0)) {
```

Reply:
I knew you would come up with this haha! I actually thought a moment about doing it this way but somehow implicit won in my mind again probably because its less chars or so.. idk 🙈 will try directly next time because i actually agree, its better the other way.. 🙂